File: ts/src/test/Exchange/base/test.marginMode.ts
import { Exchange } from "../../../../ccxt";
import testSharedMethods from './test.sharedMethods.js';

function testMarginMode (exchange: Exchange, skippedProperties: object, method: string, entry: object) {
    const format = {
        'info': {},
        'symbol': 'BTC/USDT:USDT',
        'marginMode': 'cross',
    };
    const emptyAllowedFor = [ 'symbol' ];
    testSharedMethods.assertStructure (exchange, skippedProperties, method, entry, format, emptyAllowedFor);
}

export default testMarginMode;

File: julia/Ccxt/test/validators/test.marginMode.jl
using Test
using Ccxt
function testMarginMode(exchange, skippedProperties, method, entry)

    format = Dict{Symbol, Any}(
        Symbol("info") => Dict{Symbol, Any}(),
        Symbol("symbol") => "BTC/USDT:USDT",
        Symbol("marginMode") => "cross"
    );
    emptyAllowedFor = ["symbol"];
    assertStructure(testSharedMethods, exchange, skippedProperties, method, entry, format, emptyAllowedFor);
end
